File: third-party-demo/thrift-multiple-language/thrift-pylab/com/shsz/young/thrift/UserBiz.py
# -*- coding: utf-8 -*- 

if __name__ == '__main__':
  # test with Spyder
  import os, sys
  p = os.path.split(os.path.realpath(__file__))[0]
  sys.path.append('%s/%s' % (p, '../../../..'))
  
from com.shsz.young.thrift.proto.constants import *

class UserBiz:
  def __init__(self):
    self.users = {}
  
  def store(self, user):
    print 'stored: %s' % user
    self.users[user.uid] = user
    
  def retrieve(self, uid):
    return self.users[uid]
    
if __name__ == '__main__':
  user = UserProfile(1, 'jeff', 'blahblah')
  s = UserBiz()
  s.store(user)
  print s.retrieve(user.uid)

File: third-party-demo/thrift-multiple-language/thrift-rblab/user_biz.rb
# -*- coding: utf-8 -*- 
$:.push('lib')

class UserBiz
  def initialize()
    @users = {}
  end

  def store(u)
    puts "user: #{u.inspect}"
    @users[u.uid] = u
  end

  def retrieve(uid)
    u = @users[uid]
    puts "user: #{u.inspect}"
    return u
  end
end 

if __FILE__ == $0
  $:.unshift File.dirname(__FILE__) + '/lib'
  require 'thrift'
  require 'user_constants'
  include ::UserThrift

  user = UserProfile.new()   
  user.uid = 1               
  user.name = "john李"       
  user.blurb = "blahblahhhhh"

  ub = UserBiz.new()
  ub.store(user)
  puts ub.retrieve(1).inspect
end
